src/pages/index.js

```diff
@@ -36,82 +83,176 @@ const editCardPopup = new PopupWithForm(
-  return cardElement; // Return the card's HTML element
+  return cardElement;
 }
 
-/* Event Handlers */
+// Event Handlers
+
+function handleEditAvatarSubmit(formData) {
+  editAvatarPopup.renderLoading(true);
+  const avatarUrl = formData.avatarUrl;
+  api
+    .updateProPic(avatarUrl)
+    .then((updatedUserData) => {
+      console.log(updatedUserData);
+      userInfo.setuserAvatar(updatedUserData.avatar);
+      editAvatarPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error updating avatar:", error);
+    })
+    .finally(() => {
+      editAvatarPopup.renderLoading(false, "Save");
+    });
+}
 
 function handleProfileEditSubmit(formData) {
-  userInfo.setUserInfo({
-    name: formData.title,
-    description: formData.description,
-  });
-  editCardPopup.close();
+  editCardPopup.renderLoading(true);
+  api
+    .updateProInfo(formData.title, formData.description)
+    .then((updatedUserData) => {
+      userInfo.setUserInfo({
+        name: updatedUserData.name,
+        job: updatedUserData.about,
+      });
+      editCardPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error updating profile:", error);
+    })
+    .finally(() => {
+      editCardPopup.renderLoading(false, "Save");
+    });
 }
 
 function handleAddCardSubmit(formData) {
+  newCardPopup.renderLoading(true);
   const cardData = {
     name: formData.title,
     link: formData.url,
   };
-
-  const cardElement = createCard(cardData); // Create a new card element
-  cardSection.addItem(cardElement); // Add card to the DOM
-  newCardPopup.close();
-  addCardForm.reset();
-  addFormValidator.toggleButtonState(); // Reset form button state if needed
+  api
+    .createCard(cardData)
+    .then((newCard) => {
+      const cardElement = createCard(newCard);
+      cardSection.addItem(cardElement);
+      newCardPopup.close();
+      addCardForm.reset();
+      addFormValidator.toggleButtonState();
+    })
+    .catch((error) => {
+      console.error("Error adding new card:", error);
+    })
+    .finally(() => {
+      newCardPopup.renderLoading(false, "Save");
+    });
 }
 
 function handleImageClick(data) {
   popupWithImage.open({
-    link: data.link, // URL for the image
-    name: data.name, // Name for the caption
+    link: data.link,
+    name: data.name,
   });
 }
 
-/* Event Listeners */
+function handleLikeClick(cardId, isLiked) {
+  return api
+    .likeCard(cardId, isLiked)
+    .then((updatedCard) => {
+      return updatedCard;
+    })
+    .catch((err) => {
+      console.error("Error toggling like:", err);
+      throw err;
+    });
+}
 
-profileEditBtn.addEventListener("click", () => {
-  const allData = userInfo.getUserInfo(); // Get the user info using the method
-  profileTitleInput.value = allData.name; // Set the input values with the retrieved data
-  profileDescriptionInput.value = allData.description;
-  editCardPopup.open(); // Open the edit form popup
```

Comment:
* You need to **insert** the profile data into the inputs when you open the **profile modal** to let the user **edit** the data and submit it.


You had this code previously. Please, return it back

```js


profileEditBtn.addEventListener("click", () => {
  const allData = userInfo.getUserInfo(); // Get the user info using the method
  profileTitleInput.value = allData.name; // Set the input values with the retrieved data
  profileDescriptionInput.value = allData.job;  // job is used here instead of description
  editCardPopup.open(); // Open the edit form popup
});

```

Reply:
got it thanks